From e3c0e743803fde03e65a21ca4b9031ea0c511fc9 Mon Sep 17 00:00:00 2001
From: Reto Schneider <reto.schneider@husqvarnagroup.com>
Date: Sun, 23 Jan 2022 19:46:46 +0100
Subject: [PATCH] rtl8xxxu: Raise error level of IQK failures

Despite being recoverable, RX IQ calibration failure prints a warning.
Final IQ calibration errors should be noticed even more so.

Upstream-Status: Pending
---
 drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
index 8a55f90aab6a..27c7dd38f288 100644
--- a/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
+++ b/drivers/net/wireless/realtek/rtl8xxxu/rtl8xxxu_core.c
@@ -3389,7 +3389,7 @@ static void rtl8xxxu_phy_iqcalibrate(struct rtl8xxxu_priv *priv,
 			break;
 		} else if (i == (retry - 1) && path_a_ok == 0x01) {
 			/* TX IQK OK */
-			dev_dbg(dev, "%s: Path A IQK Only Tx Success!!\n",
+			dev_err(dev, "%s: Path A IQK only TX success!\n",
 				__func__);
 
 			val32 = rtl8xxxu_read32(priv,
@@ -3402,7 +3402,7 @@ static void rtl8xxxu_phy_iqcalibrate(struct rtl8xxxu_priv *priv,
 	}
 
 	if (!path_a_ok)
-		dev_dbg(dev, "%s: Path A IQK failed!\n", __func__);
+		dev_err(dev, "%s: Path A IQK failed!\n", __func__);
 
 	if (priv->tx_paths > 1) {
 		/*
@@ -3429,6 +3429,9 @@ static void rtl8xxxu_phy_iqcalibrate(struct rtl8xxxu_priv *priv,
 				break;
 			} else if (i == (retry - 1) && path_b_ok == 0x01) {
 				/* TX IQK OK */
+				dev_err(dev,
+					"%s: Path B IQK only TX success!\n",
+					__func__);
 				val32 = rtl8xxxu_read32(priv, REG_TX_POWER_BEFORE_IQK_B);
 				result[t][4] = (val32 >> 16) & 0x3ff;
 				val32 = rtl8xxxu_read32(priv, REG_TX_POWER_AFTER_IQK_B);
@@ -3437,7 +3440,7 @@ static void rtl8xxxu_phy_iqcalibrate(struct rtl8xxxu_priv *priv,
 		}
 
 		if (!path_b_ok)
-			dev_dbg(dev, "%s: Path B IQK failed!\n", __func__);
+			dev_err(dev, "%s: Path B IQK failed!\n", __func__);
 	}
 
 	/* Back to BB mode, load original value */
-- 
2.43.0

